backends/imgui_impl_vulkan.cpp

```diff
@@ -1048,10 +1064,11 @@ bool    ImGui_ImplVulkan_LoadFunctions(PFN_vkVoidFunction(*loader_func)(const ch
         return false;
     IMGUI_VULKAN_FUNC_MAP(IMGUI_VULKAN_FUNC_LOAD)
 #undef IMGUI_VULKAN_FUNC_LOAD
-#else
-    IM_UNUSED(loader_func);
-    IM_UNUSED(user_data);
 #endif
+
+	imgui_vkCmdBeginRenderingKHR = reinterpret_cast<PFN_vkCmdBeginRenderingKHR>(loader_func("vkCmdBeginRenderingKHR", user_data));
+	imgui_vkCmdEndRenderingKHR = reinterpret_cast<PFN_vkCmdEndRenderingKHR>(loader_func("vkCmdEndRenderingKHR", user_data));
+
```

Comment:
This way the user is required to call `ImGui_ImplVulkan_LoadFunctions()` even in the case it wouldn't call it otherwise (i.e. when linking with Vulkan and using prototypes).

For the no VK_NO_PROTOTYPES case I suggest you delay initializing those to `ImGui_ImplVulkan_Init()` and either use `vkGetInstanceProcAddr()` directly there (preferred in this case) or set things up to call `ImGui_ImplVulkan_LoadFunctions()` from there.

Reply:
Oh true, I thought it was being called by default from the Init function. I tested the glfw/Vulkan example and it ran fine so I assumed it was working :) Will fix as soon as I get home.

Reply:
(Sorry completely forgot about this). I've now gone ahead and made it so that the function pointers are *always* loaded in the Init function if `UseDynamicRendering` is true.

Reply:
See my suggestion in the other comment for handling LoadFunctions. If user gives a loader callback, we want to always use it, otherwise would be surprising and we would have to document.

What is the user provided loader used for?  One use is some applications will not link at build time with the Vulkan Loader, instead will manually find and open the library and get the vkGetInstanceProcAddr by themselves.  The custom function allow ImGui to grab all the functions it needs through that pointer.

As for the comment, I think is just pointing that if the vkGetInstanceProcAddr is available when linking during build, it is very likely the other core functions will be available.  This could be tweaked but I suggest proposing the tweak in a separate PR.